### PR TITLE
ovl_Obj_Switch OK and documented

### PR DESCRIPTION
**Pull request by Dragorn421**
I find defines like this one make the code a lot more readable
```c
#define SWITCH_FLAG (this->dyna.actor.params >> 8 & 0x3F)
```
they are `#undef`ed at the end of the .c file.

Hopefully empty lines between "sub-systems" like Floor/Eye/Crystal are okay in the prototype list.

The `extern` symbols cannot go after data because data references them.

I didn't name all enum types because I'm not sure what the behaviors are. Will be a lot easier once collision stuff is decompiled and flags are documented.

The `union`s at the end of the `ObjSwitch` structure look janky, suggestions welcome.

Criticism about style, names and abundance of comments is welcome as always.

### CODE REVIEW

**Comment by fig02** [comment]
defines for flags and other parts of params are definitely planned to happen but are not a part of the decompilation step, its something that will happen on a larger scale later

**Comment by Dragorn421** [comment]
Sure, should I remove them, adding them in was a quick ctrl+H and helped me a lot documenting process but it can go just as easily now

**Review by fig02 — APPROVED** [review]
Implemented suggested changes and my own review on top.

File: src/overlays/actors/ovl_Obj_Switch/z_obj_switch.c [hunk 1]
```diff
@@ -27,89 +68,692 @@ const ActorInit Obj_Switch_InitVars = {
...
+
+void ObjSwitch_InitJntSphCollider(ObjSwitch* this, GlobalContext* globalCtx, ColliderJntSphInit* colliderJntSphInit) {
+    ColliderJntSph* colliderJntSph = &this->jntSph.col;
+
+    Collider_InitJntSph(globalCtx, colliderJntSph);
+    Collider_SetJntSph(globalCtx, colliderJntSph, &this->dyna.actor, colliderJntSphInit, this->jntSph.items);
+    func_800D1694(this->dyna.actor.posRot.pos.x,
+                  this->dyna.actor.posRot.pos.y + this->dyna.actor.shape.unk_08 * this->dyna.actor.scale.y,
+                  this->dyna.actor.posRot.pos.z, &this->dyna.actor.shape.rot);
+    Matrix_Scale(this->dyna.actor.scale.x, this->dyna.actor.scale.y, this->dyna.actor.scale.z, 1);
```
Comment (Roman971) [inline src/overlays/actors/ovl_Obj_Switch/z_obj_switch.c:163]:
```suggestion
    Matrix_Scale(this->dyna.actor.scale.x, this->dyna.actor.scale.y, this->dyna.actor.scale.z, MTXMODE_APPLY);
```

File: src/overlays/actors/ovl_Obj_Switch/z_obj_switch.c [hunk 2]
```diff
@@ -27,89 +68,692 @@ const ActorInit Obj_Switch_InitVars = {
...
+}
+
+void ObjSwitch_FloorUpInit(ObjSwitch* this) {
+    this->dyna.actor.scale.y = 33.0f / 200.0f;
+    this->actionFunc = ObjSwitch_FloorUp;
+}
+
+void ObjSwitch_FloorUp(ObjSwitch* this, GlobalContext* globalCtx) {
+    if ((this->dyna.actor.params & 7) == OBJSWITCH_TYPE_FLOOR_RUSTY) {
+        if ((this->tris.col.base.acFlags & 2) != 0) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_Obj_Switch/z_obj_switch.c:330]:
```suggestion
        if (this->tris.col.base.acFlags & 2) {
```

**Comment by fig02** [comment]
changes made

**Review by Roman971 — APPROVED** [review]

**Pull request merged**